### ISSUE TITLE
Gauge visualization using 'null' fill values leads to impression that something is broken 

### ISSUE THREAD

**Issue by karel-rehor** [influxdata/chronograf#3591]
build **nightly build** 

build time **n201806060834**

browser **Chrome 66**

```
$ chronograf --version
2018/06/07 10:45:08 Chronograf 1.2.0~n201806060834 (git: dd8dfc56c9f965eb2dba59800bf164c46884ed1b)
```

**Use Case**

1. Create a new dashboard
2. Add Data to the dashboard using the builder. telegraf.autogen > cpu > usage_idle. A line graph appears in the preview window. 
3. Click visualizations
4. Click gauge 

The gauge seems to be stuck at 0 even though the line graph and other visualizations show no '0' value data is collected.  It seems the gauge is broken. 

**Expected Behaviour** 

That the gauge would reflect the lead end of the line graph curve. 

**Further investigation** 

The table view provides a clue as to what is happening.  The generated query is as follows: 

```
SELECT mean("usage_idle") AS "mean_usage_idle" FROM "telegraf"."autogen"."cpu" WHERE time > :dashboardTime: GROUP BY time(:interval:) FILL(null)
```
The table view reveals that with `time(:interval:)` as a GROUP BY argument and with FILL(NULL) the last value is often "null", which the gauge translates to 0.  

**Suggestion** 

Since some users will not be familiar enough with InfluxQL and the underlying stack to intuit the cause of this problen, perhaps it would be better if the gauge ignored "fill" values set to "null" and instead look for the last valid numerical value in the stream of datapoints. 

**Screenshots**
![influxqlwithgauge01](https://user-images.githubusercontent.com/25881301/41089080-48cc998e-6a41-11e8-9257-94ee06c7f81c.png)
![influxqlwithgauge02](https://user-images.githubusercontent.com/25881301/41089081-48ef7846-6a41-11e8-89fb-5f958b389f97.png)
![influxqlwithgauge03](https://user-images.githubusercontent.com/25881301/41089082-490fe9aa-6a41-11e8-8c71-41cf28c86081.png)
![influxqlwithgauge04](https://user-images.githubusercontent.com/25881301/41089083-49304592-6a41-11e8-8eab-8b0d80dcd31e.png)
![influxqlwithgauge06](https://user-images.githubusercontent.com/25881301/41089091-4daef01e-6a41-11e8-8920-51d5adb91e1a.png)

**Comment by russorat** [comment]
the gauge really only works with `fill(previous)` enabled. could you try your queries with that and see what the results are? we do need to do a better job of populating that for the user.

**Comment by karel-rehor** [comment]
Yes.  I've tried queries with `fill(previous)` and the gauge works.  

However, for users new to TICK stack and chronograf who may try to simply click through telegraf.autogen > cpu > usage_idle (or something similar which defaults to a mean measurement taken from a data window, with default `fill(null)`) they may come to a similar conclusion. 

There is no mention of specific fill type for gauge in the documentation, although with a little reflection after a short while the cause of the problem is apparent. 

https://docs.influxdata.com/chronograf/v1.5/guides/visualization-types/#gauge

Some tool-tip or hints in the UI, for example 'works best with fill(previous)', when choosing the gauge visualization, might be beneficial to new users.  Also could be mentioned in the documentation directly under the visualization-types/#gauge topic.  When I saw the gauge stuck at 0, my first inclination was to open the documentation. 

A more complicated approach would be for the gauge control to reject 'null' values, instead of casting them to '0', and looking for the first numerical (type) value.
